shell/common/shell.cc

```diff
@@ -1223,7 +1223,20 @@ void Shell::OnEngineHandlePlatformMessage(
   }
 
   if (platform_message_handler_) {
-    if (route_messages_through_platform_thread_) {
+    if (route_messages_through_platform_thread_ &&
+        !platform_message_handler_
+             ->DoesHandlePlatformMessageOnPlatformThread()) {
+#if _WIN32
+      // On Windows capturing a TaskRunner with a TaskRunner will cause an
+      // uncaught exception in process shutdown because of the deletion order of
+      // global variables. See also
+      // https://github.com/flutter/flutter/issues/111575.
+      // This won't be an issue until Windows supports background platform
+      // channels. Then this can potentially be addressed by capturing a
+      // weak_ptr to an object that retains the ui TaskRunner, instead of the
+      // TaskRunner directly.
+      FML_DCHECK(false);
+#endif
```

Comment:
Typically I wouldn't do this but this situation is pretty difficult to debug and I lost a couple days on it so I don't want to lose it.  I have another hypothetical refactor that would make it cleaner until we implement background platform channels but I'm not sure if I'll get to finish it (https://github.com/flutter/engine/pull/36157).